### PR TITLE
Add test object parameter reference

### PR DESCRIPTION
**Pull request by kingo55**
Hey guys,

I documented the test object parameters. I'm not 100% sure about the functionality for shared JS/CSS though.

Does this seem accurate?:

## Test object: Root level

Parameter | Description
--|--
**css** <br> Type: *string* <br> *Optional* | Shared CSS for the test object that is applied to all recipes. <br>`JS format`: Expects a JS string of CSS styles. <br>`YAML format`: Expects a relative path to a CSS file.
**divertTo** <br> Type: *string* <br> *Optional* | Enable divert mode to send all eligible traffic into a particular recipe's key.
**id** <br> Type: *string* <br> *Required* | A canonical test object ID by which subjects' assignments are recorded against. 
**js** <br> Type: *function* <br> *Required* | Shared JS for the test object that is applied to all recipes. <br>`JS format`: Expects a valid JavaScript function on this key. <br>`YAML format`: Expects a relative path to a JS file with that function.
**name** <br> Type: *string* <br> *Required* | The name of the test object that's useful in tracking.
**recipes** <br> Type: *object* <br> *Required* | An object containing the definition of available recipes.
**sampleRate** <br> Type: *number* <br> *Required* | The percentage of traffic to assign into the experiment between 0 and 1 (where 1 is 100%)
**state** <br> Type: *string* <br> *Required* | A test object's state. Can be either: <br> `live` - in the container, accepting traffic into the experiment <br>`staging` - built into the container but not accepting traffic without a preview URL <br>`inactive` - not parsed or built into the container
**trigger** <br> Type: *function* <br> *Required* | A JavaScript function executed as soon as the test object is loaded and can be used to conditionally activate an experiment. <br>`JS format`: Expects a valid JavaScript function on this key. <br>`YAML format`: Expects a relative path to a JS file with that function.

localhost_3001_docs_js-delivery-test-object

#3

### CODE REVIEW

File: docs/js-delivery-test-object.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+
+## Test object: Root level
+
+Parameter | Description
+--|--
+**css** <br> Type: *string* <br> *Optional* | Shared CSS for the test object that is applied to all recipes. <br>`JS format`: Expects a JS string of CSS styles. <br>`YAML format`: Expects a relative path to a CSS file.
+**divertTo** <br> Type: *string* <br> *Optional* | Enable divert mode to send all eligible traffic into a particular recipe's key.
+**id** <br> Type: *string* <br> *Required* | A canonical test object ID by which subjects' assignments are recorded against. 
+**js** <br> Type: *function* <br> *Required* | Shared JS for the test object that is applied to all recipes. <br>`JS format`: Expects a valid JavaScript function on this key. <br>`YAML format`: Expects a relative path to a JS file with that function.
```
Comment (allmywant) [inline docs/js-delivery-test-object.md]: The shared JS can be a function or an JS object:
js: function shared() { do something }
js: {
    propA: 'a',
    fnA: function() { },
    fnB: function() { }
}

Comment (kingo55) [reply docs/js-delivery-test-object.md]: Gotcha... and we can represent this in the YAML format by a regular JS object in a file?

Comment (allmywant) [reply docs/js-delivery-test-object.md]: Yes we can.

File: docs/js-delivery-test-object.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+**name** <br> Type: *string* <br> *Required* | The name of the test object that's useful in tracking.
+**recipes** <br> Type: *object* <br> *Required* | An object containing the definition of available recipes.
```
Comment (dapperdrop) [inline docs/js-delivery-test-object.md]: Add a link on "object" to the Recipe Object section
Likewise add a link on "definition" to the Receipe objects sub-level section

File: docs/js-delivery-test-object.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+Parameter | Description
+--|--
+**{{recipeId}}** <br> Type: *string* <br> *Required* | The canonical key that references a test object within the cookies and preview URLs. E.g. A recipe ID of `a` would be accesible through a preview URL like: `https://www.example.com/?mojito_w12=a`
+
+## Recipe objects sub-level
+
+Parameter | Description
+--|--
+**css** <br> Type: *string* <br> *Optional* | CSS that is applied as soon as the subject is bucketed. <br>`JS format`: Expects a JS string of CSS styles. <br>`YAML format`: Expects a relative path to a CSS file.
+**js** <br> Type: *function* <br> *Optional* | A JavaScript function that runs as soon as a subject is assigned into the test. <br>`JS format`: Expects a valid JavaScript function on this key. <br>`YAML format`: Expects a relative path to a JS file with that function.
```
Comment (dapperdrop) [inline docs/js-delivery-test-object.md]: The description here should match the experiment shared js, i think?

Comment (kingo55) [reply docs/js-delivery-test-object.md]: @allmywant - can recipe JS keys accept ```function``` and ```object``` types, too?

Comment (allmywant) [reply docs/js-delivery-test-object.md]: Recipe JS keys accept function only.

**Pull request merged**